### PR TITLE
Add test that simulates 16k vault creations + transfers 

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

Note this is just for @JosDenmark @ArturGajowy and anyone else who is investigating the full genesis ceremony execution. This test is a simplified version of the full genesis ceremony execution that brings forth all of the main issues: how to get 16k vault creations + transfers done in a reasonable amount of time.

Instructions for reproducing flamegraphs

0. Start the test with `sbt casper/testOnly *RholangBuildTest*` (note there are two subtests so wait for the first to finish)
1. `jps` (to find PID - for me it is always named ForkMain) 
2. `./profiler.sh -d 100 -f /tmp/flamegraph.svg <PID>`

Flamegraphs

After commit 3420905 (see https://transfer.sh/N8nHM/flamegraph.svg) taken ~2 minutes after test starts. Around 1 vault creation/sec.

Screen Shot 2019-07-24 at 2 07 38 AM

After commit bbd296b (see https://transfer.sh/88oI1/flamegraph.svg) taken ~2 minutes after test starts. Around 5 vault creations/sec.

Screen Shot 2019-07-24 at 2 19 07 AM

After commit bbd296b (see https://transfer.sh/broUv/flamegraph.svg) taken ~4 hour after test starts. Around 0.5 vault creations/sec.

Screen Shot 2019-07-24 at 7 29 09 AM

After commit 1755631 (see https://transfer.sh/xU039/flamegraph.svg) taken ~10 minutes after test starts. Around 1.5 vault creations/sec.

Screen Shot 2019-07-24 at 11 47 50 PM

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Review by ArturGajowy — CHANGES_REQUESTED** [review]
There's no way we're merging this with the 300k lines of code that could be generated on the fly

**Comment by KentShikama** [comment]
If it wasn't clear by the description, this is not meant to be merged

File: models/src/main/scala/coop/rchain/models/rholang/sorter/ExprSortMatcher.scala [hunk 1]
```diff
@@ -259,7 +260,7 @@ private[sorter] object ExprSortMatcher extends Sortable[Expr] {
       case gs: GString => ScoredTerm(e, Node(Score.STRING, Leaf(gs.value))).pure[F]
       case gu: GUri    => ScoredTerm(e, Node(Score.URI, Leaf(gu.value))).pure[F]
       case GByteArray(ba) =>
-        ScoredTerm(e, Node(Score.EBYTEARR, Leaf(ba.toStringUtf8))).pure[F]
+        ScoredTerm(e, Node(Score.EBYTEARR, Leaf(Base16.encode(ba.toByteArray)))).pure[F]
```
Comment (ArturGajowy) [inline models/src/main/scala/coop/rchain/models/rholang/sorter/ExprSortMatcher.scala:263]: base16.encode is a hellishly slow way to sort. We should rather use the byte array directly in the leaf, nad make sure that we sort Leaf(byteArray) using the array elements

File: models/src/main/scala/coop/rchain/models/SortedParMap.scala [hunk 2]
```diff
@@ -1,59 +1,63 @@
...
   def --(keys: GenTraversableOnce[Par]): SortedParMap =
-    SortedParMap(keys.foldLeft(sortedMap) { (map, kv) =>
-      map - sort(kv)
+    SortedParMap(keys.foldLeft(ps) { (map, key) =>
+      map - sort(key)
     })
 
-  def apply(par: Par): Par = sortedMap(sort(par))
-
-  def contains(par: Par): Boolean = sortedMap.contains(sort(par))
```
Comment (ArturGajowy) [inline models/src/main/scala/coop/rchain/models/SortedParMap.scala:29]: as mentioned in discord, lookups would be more efficient if we kept both a hashmap and a treemap, and did the lookups inside the hashMap

File: models/src/main/scala/coop/rchain/models/SortedParMap.scala [hunk 3]
```diff
@@ -44,15 +44,25 @@ final case class SortedParMap(ps: TreeMap[Par, Par]) extends Iterable[(Par, Par)
 }
 
 object SortedParMap {
+  @SuppressWarnings(Array("org.wartremover.warts.Var", "org.wartremover.warts.Null"))
```
Comment (JosephDenman) [inline models/src/main/scala/coop/rchain/models/SortedParMap.scala:47]: @ArturGajowy @KentShikama This is why I used `TreeMap[ScoredTerm[Par], Par]`. No cache, no resorting.

**Pull request closed without merging**